drivers/mpu9250/MPU9250.cpp

```diff
@@ -343,20 +344,25 @@ void MPU9250::reset_fifo()
 
 	int result;
 
-	if (_mag_enabled) {
-		result = _writeReg(MPUREG_USER_CTRL,
-				   BITS_USER_CTRL_FIFO_RST |
-				   BITS_USER_CTRL_FIFO_EN |
-				   BITS_USER_CTRL_I2C_MST_EN);
+	result = _modifyReg(MPUREG_USER_CTRL,
+			    0,
+			    BITS_USER_CTRL_FIFO_RST |
+			    BITS_USER_CTRL_FIFO_EN);
 
-	} else {
-		result = _writeReg(MPUREG_USER_CTRL,
-				   BITS_USER_CTRL_FIFO_RST |
-				   BITS_USER_CTRL_FIFO_EN);
+	if (result != 0) {
+		DF_LOG_ERR("FIFO reset failed");
 	}
+}
+
+void MPU9250::clear_int_status()
+{
+	int result;
+	uint8_t int_status = 0;
+
+	result = _readReg(MPUREG_INT_STATUS, int_status);
```

Comment:
I guess this is all it needed, awesome! Thanks a lot, I was debugging this for several days and couldn't figure it out.